src/irmin/store.ml

```diff
@@ -1153,9 +1153,9 @@ module Make (B : Backend.S) = struct
         let current, current_depth = Heap.pop_minimum heap in
         let parents = Commit.parents current in
         let tree = Commit.tree current in
-        let* current_value = Tree.find tree key in
+        let* current_tree = Tree.find_tree tree key in
```

Comment:
I think this could also be optimized away since, except for the very first call, we already fetched the `current_tree` on the previous recursive call on line 1175 below :) (but we can do this in a follow-up PR if it's unclear how to)